HLTrigger/JetMET/plugins/HLTJetHFCleaner.h

```diff
@@ -0,0 +1,219 @@
+      sigmaEtaEtaToken_(consumes(iConfig.getParameter<edm::InputTag>("sigmaEtaEta"))),
+      sigmaPhiPhiToken_(consumes(iConfig.getParameter<edm::InputTag>("sigmaPhiPhi"))),
+      centralEtaStripSizeToken_(consumes(iConfig.getParameter<edm::InputTag>("centralEtaStripSize"))),
+      jetPtThresh_(iConfig.getParameter<double>("jetPtThresh")),
+      dphiJetMetThresh_(iConfig.getParameter<double>("dphiJetMetThresh")),
+      jetEtaMin_(iConfig.getParameter<double>("jetEtaMin")),
+      jetEtaMax_(iConfig.getParameter<double>("jetEtaMax")),
+      sigmaEtaPhiDiffThresh_(iConfig.getParameter<double>("sigmaEtaPhiDiffThresh")),
+      centralEtaStripSizeThresh_(iConfig.getParameter<int>("centralEtaStripSizeThresh")),
+      cornerCutSigmaEtaEta_(iConfig.getParameter<double>("cornerCutSigmaEtaEta")),
+      cornerCutSigmaPhiPhi_(iConfig.getParameter<double>("cornerCutSigmaPhiPhi")),
+      applyCornerCut_(iConfig.getParameter<bool>("applyCornerCut")),
+      applySigmaEtaPhiCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCut")),
+      applyStripSizeCut_(iConfig.getParameter<bool>("applyStripSizeCut")) {
+  produces<JetCollection>();
+}
+
+template <typename JetType>
+void HLTJetHFCleaner<JetType>::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  auto const jets = iEvent.getHandle(jetToken_);
+  auto const mets = iEvent.getHandle(metToken_);
+
+  auto const& sigmaEtaEtas = iEvent.get(sigmaEtaEtaToken_);
+  auto const& sigmaPhiPhis = iEvent.get(sigmaPhiPhiToken_);
+  auto const& centralEtaStripSizes = iEvent.get(centralEtaStripSizeToken_);
+
+  // The set of cleaned jets
+  auto cleaned_jets = std::make_unique<JetCollection>();
+  cleaned_jets->reserve(jets->size());
+
+  // Check if length of the MET vector is 1, throw exception otherwise
+  if (mets->size() != 1) {
+    throw cms::Exception("ComparisonFailure", "Length of MET vector is different than 1");
+  }
+  auto const met_phi = mets->front().phi();
+
+  // Loop over the jets and do cleaning
+  for (uint ijet = 0; ijet < jets->size(); ijet++) {
+    auto const jet = (*jets)[ijet];
+    auto const withinEtaRange = (std::abs(jet.eta()) < jetEtaMax_) and (std::abs(jet.eta()) > jetEtaMin_);
+    // Cuts are applied to jets that are back to back with MET
+    if (reco::deltaPhi(jet.phi(), met_phi) <= dphiJetMetThresh_) {
+      cleaned_jets->emplace_back(jet);
+      continue;
+    }
+    // If the jet is outside the range that we're interested in,
+    // append it to the cleaned jets collection and continue
+    if ((jet.pt() < jetPtThresh_) or (!withinEtaRange)) {
+      cleaned_jets->emplace_back(jet);
+      continue;
+    }
+
+    JetRef const jetRef(jets, ijet);
+
+    // Extract HF shape variables from the ValueMap inputs
+    auto const sigmaEtaEta = sigmaEtaEtas[jetRef];
+    auto const sigmaPhiPhi = sigmaPhiPhis[jetRef];
+    auto const centralEtaStripSize = centralEtaStripSizes[jetRef];
```

Comment:
By the way, these VMap calls are not always needed (as their use depends on flags like `applySigmaEtaPhiCut_`). We can move these line inside the relevant `if` blocks. Even the `consume` calls of the VMaps could be made conditional (meaning, for example, not calling consumes for VMaps that won't be used in ::produce, based on the value of flags).

Reply:
Ah sure I can update that. So within the constructor, I can only call the `consume` calls depending on values of corresponding boolean flags. And then in `produce` method, retrieve them only if the flags are `true`.

Reply:
I tried to address this in [a1fc8c9](https://github.com/cms-sw/cmssw/pull/37539/commits/a1fc8c97ce8e96cda540d78f8f7d138bb58aa9b3). I moved the `get` methods in corresponding if blocks, but I didn't touch the constructor because of compiler complications about which members getting initialized first (if one of the flags is `False` and the other one is `True`, the order of initializations gets swapped). Let me know what you think and if you want more changes to this, thanks!

Reply:
@alpakpinar , thanks for trying. I realise this is not entirely trivial, and I don't want to delay this PR for this (i might decide to address this technical point in the future). [by the way, we would not move the `.get` calls inside the loop]

So, I retract the comment, and suggest to undo a1fc8c97ce8e96cda540d78f8f7d138bb58aa9b3. I left one more comment below, which should be my final one.

One more thing: in your next update, please squash all commits into 1, and force-push to the same branch. The result will be a PR with just 1 commit (the discussion remains on GitHub, and on git we dont need/want to keep centrally all these intermediate commits).

Reply:
Ah you're right sorry, I shouldn't have put the `get` call inside the `for` loop. Okay then, thanks so much and I'll add a `revert` commit for this one. After I've addressed your other comments, I'll squash all commits into 1 and force-push as you said.